### DISCUSSION TITLE
how to attach javagents while starting up keycloak on quarkus

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#29396] [Archive]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

JAVA_OPTS_APPEND can be used for passing on -D params during start up,
how can i attach agents (say if i want to attach a profiler agent or otel agent) what should i do

### Version

24.x

### Regression

- [ ] The issue is a regression

### Expected behavior

should have provisions to conditionaly add agents during startup

### Actual behavior

not defined

### How to Reproduce?

appears that JAVA_OPTS_APPEND does not honor -javaagent

say wile starting up i want to do 
-javaagent:codeguru-profiler-java-agent-standalone-1.2.3.jar="profilingGroupName:keycloak,heapSummaryEnabled:true"
not sure how to get this done without modifying kc.sh

**Comment by shawkins** [comment]
Moved this to a discussion. 

JAVA_OPTS_APPEND is already recognized as an ENV variable, if I do something like:

JAVA_OPTS_APPEND=-javaagent:codeguru-profiler-java-agent-standalone-1.2.3.jar="profilingGroupName:keycloak,heapSummaryEnabled:true" ./kc.sh start-dev

I can see on the console that the property is being used:

Appending additional Java properties to JAVA_OPTS

And my vm fails to start because I don't actually have that codeguru-profiler-java-agent-standalone-1.2.3.jar

Why do you expect it to be handled as a system property?

**Comment by kkcmadhu** [comment]
well, i was able to acheive it iwht java_opts_append something like this

        - name: "AWS_CODEGURU_PROFILER_HEAP_SUMMARY_ENABLED"
          value: "true"        
        - name: "AWS_CODEGURU_PROFILER_GROUP_NAME"
          value: "keycloak-perf-test"        
        - name: "JAVA_OPTS_APPEND"
          value: '-javaagent:"/path/to/myprofilerdirectory/profiler-java-agent.jar"'
